src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForLoopStatement.cs

```diff
@@ -0,0 +1,2041 @@
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public partial class IOperationTests : SemanticModelTestBase
+    {
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForLoopStatement_ForLoop()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        int x = 3;
+        /*<bind>*/for (int i = 0; i < 3; i = i + 1)
+        {
+            x = x * 3;
+        }/*</bind>*/
+        System.Console.Write(""{0}"", x);
+    }
+}
+";
+            string expectedOperationTree = @"
+IForLoopStatement (LoopKind.For) (OperationKind.LoopStatement)
+  Condition: IBinaryOperatorExpression (BinaryOperationKind.IntegerLessThan) (OperationKind.BinaryOperatorExpression, Type: System.Boolean)
+      Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+      Right: ILiteralExpression (Text: 3) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3)
+  Local_1: System.Int32 i
+  Before: IVariableDeclarationStatement (1 variables) (OperationKind.VariableDeclarationStatement)
+      IVariableDeclaration: System.Int32 i (OperationKind.VariableDeclaration)
+        Initializer: ILiteralExpression (Text: 0) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0)
+  AtLoopBottom: IExpressionStatement (OperationKind.ExpressionStatement)
+      IAssignmentExpression (OperationKind.AssignmentExpression, Type: System.Int32)
+        Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: i (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 1) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 1)
+  IBlockStatement (1 statements) (OperationKind.BlockStatement)
+    IExpressionStatement (OperationKind.ExpressionStatement)
+      IAssignmentExpression (OperationKind.AssignmentExpression, Type: System.Int32)
+        Left: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32)
+        Right: IBinaryOperatorExpression (BinaryOperationKind.IntegerMultiply) (OperationKind.BinaryOperatorExpression, Type: System.Int32)
+            Left: ILocalReferenceExpression: x (OperationKind.LocalReferenceExpression, Type: System.Int32)
+            Right: ILiteralExpression (Text: 3) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 3)
+";
+            VerifyOperationTreeForTest<ForStatementSyntax>(source, expectedOperationTree);
+        }
+
+        [Fact, WorkItem(17602, "https://github.com/dotnet/roslyn/issues/17602")]
+        public void IForLoopStatement_ExecuteOnce()
```

Comment:
Can you make the test names more descriptive of what the test actually tests, instead of what the test code does? `ExecutesOnce` isn't very descriptive when trying to chase down a failure